core_contracts/loans/loans/loans.py

```diff
@@ -724,23 +756,83 @@ def retireRedeem(self, _symbol: str, _sicx_from_lenders: int) -> None:
 
             remaining_supply -= user_debt
         price = asset.priceInLoop()
-        self.AssetRetired(self.msg.sender, _symbol, _redeemed, price, _redeemed,
+        self.AssetRetired(self.msg.sender, _symbol, _redeemed, price,
                           total_batch_debt, str(redeemed_dict))
 
+    @external
+    @only_rebalance
+    def generateBnusd(self, _tokens_to_retire: int) -> None:
+        """
+        This function will  add off debt to a batch of
+        borrowers proportionately along with their collateral.
+        :param _tokens_to_retire: Max bnUSD token to retire.
+        :type _tokens_to_retire: int
+        """
+        _symbol = "sICX"
+        _from = self._rebalance.get()
+        asset = self._assets[_symbol]
+
+        if not (asset and asset.is_active()):
+            revert(f'{TAG}: {_symbol} is not an active asset.')
+
+        price = asset.priceInLoop()
+        batch_size = self._redeem_batch.get()
+        borrowers = self._assets['bnUSD'].get_borrowers()
+        node_id = borrowers.get_head_id()
+        total_batch_debt: int = 0
+        positions_dict = {}
+        for _ in range(min(batch_size, len(borrowers))):
+            user_debt = borrowers.node_value(node_id)
+            positions_dict[node_id] = user_debt
+            total_batch_debt += user_debt
+            borrowers.move_head_to_tail()
+            node_id = borrowers.get_head_id()
+        borrowers.serialize()
+
+        bnusd_to_retire = min(_tokens_to_retire * POINTS, self._max_bnusd_retire.get() * POINTS,
```

Comment:
I think we need to correct our terminology here. This is the opposite of retiring bnUSD, which referred to taking the bnUSD out of circulation and burning it. Here we are minting it and selling it. I think the common thing between the forward and reverse rebalancing is that we are selling on the DEX, so I would suggest this variable be named bnusd_to_sell. the sicx that we sell in the forward case would have a similar name for the variable. It is not being retired either, just sold.